packages/common-all/src/dnode.ts

```diff
@@ -793,7 +807,17 @@ export class NoteUtils {
   }
 
   static serializeMeta(props: NoteProps) {
-    const builtinProps = _.pick(props, [
+    // Remove all undefined values, because they cause `matter` to fail serializing them
+    // @ts-ignore
```

Comment:
instead of doing a ts-ignore, can we declare it a partial here and then create a validation type narrowing function `validateNoteProps(noteprops: Partial<NoteProps>)` that validates we have the necessary fields?